### ISSUE TITLE
Add Social Media Icon Links Pattern

### ISSUE THREAD

**Issue by kaitlinbolling** [alleyinteractive/accessibility-patterns#19]
### Description

A **social media links** pattern should be added to the Accessibility Patterns Library and documented.
### Acceptance Criteria

See [Contributing Guidelines](https://github.com/alleyinteractive/accessibility-patterns/blob/main/CONTRIBUTING.md)

- A **social media links** component has been added to the pattern library. The component should illustrate best practices for icon links that represent various social media networks (Facebook, X, Instagram, YouTube, etc.).
- The component passes all WCAG AA specifications (see Testing).
- The component pattern(s) are thoroughly documented, including but not limited to the following sections:
  - **Title**: Pattern title, along with other names that may be used. 
  - **Summary**: An overview of what the pattern is and how it is commonly used. 
  - **Best Practices**: Some DOs and DON'Ts when implementing this pattern. 
  - **Accessibility Considerations**: Any specific accessibility considerations or edge cases to account for. 
  - **Examples**: Examples of markup used to build the pattern.

### Testing

Each pattern added to the library should be tested thoroughly to ensure it meets WCAG AA standards.

- **Automated Testing**: Use Google Lighthouse and the SiteImprove Accessibility Checker Chrome extension (or another tool of your choice) to ensure the component passes all criteria.
- **Manual Testing**: Conduct manual testing via keyboard and screen reader (desktop and mobile) to ensure the component is accessible and user-friendly.

**Testing Tips**

The pattern library includes the [a11y addon](https://storybook.js.org/addons/@storybook/addon-a11y) that adds an "Accessibility" tab to each component's canvas and performs automated accessibility testing. Use this to ensure there are no violations.

When running Storybook locally, you can view your component in a standalone window by clicking the "Open canvans in a new tab" icon button in the top right. You can use this window to test your component in isolation.

### Use Case

Social media links are commonly used in the footer of our clients' websites, or on individual author pages. This pattern should be added to the library and documented so that everyone at Alley can reference the pattern and create accessible components within their projects.
